Restructure the "Evaluating models" documentation page and rename this page to "Testing models" (#5457)
RasaHQ/rasa: docs/user-guide/testing-your-assistant.rst

   If you are looking to tune the hyperparameters of your NLU model,
   check out this `tutorial <https://blog.rasa.com/rasa-nlu-in-depth-part-3-hyperparameters/>`_.

.. _end_to_end_testing:

End-to-End Testing
------------------

> Rasa Open Source lets you test dialogues end-to-end, by running through

Reviewer: ```suggestion
Rasa Open Source lets you test dialogues end-to-end by running through
```

> conversation tests and making sure that both NLU and Core make correct predictions.

Reviewer: ```suggestion
test conversations and making sure that both NLU and Core make correct predictions.
```

        * inform: [afghan](cuisine) food
            - form: restaurant_form
            - form{"name": null}
            - utter_slots_values
        * thankyou: thanks
            - utter_noworries

 .. group-tab:: Forms Unhappy Path

    .. code-block:: story

        ## Testing a conversation with a form and unexpected user input
        * greet: hi
            - utter_greet
        * request_restaurant: im looking for a restaurant
            - restaurant_form
            - form{"name": "restaurant_form"}
        <!-- The user sends a message which should not be handled by the form. -->
        * chitchat: can you share your boss with me?
            - utter_chitchat
            - restaurant_form
            - form{"name": null}
            - utter_slots_values
        * thankyou: thanks
            - utter_noworries

By default Rasa Open Source saves conversation tests to ``tests/conversation_tests.md``.
You can test your assistant against them by running:

.. code-block:: bash

  $ rasa test

.. note::

  :ref:`custom-actions` are not executed as part of end-to-end tests. If your custom
  actions append any events to the tracker, this has to be reflected in your end-to-end

>   tests, e.g. by adding ``slot`` events to your end-to-end story.

Reviewer: ```suggestion
  tests (e.g. by adding ``slot`` events to your end-to-end story).
```
